WooCommerce/UITestsFoundation/Screens/ExternalAppScreen.swift

```diff
@@ -0,0 +1,34 @@
+import XCTest
+
+public final class ExternalAppScreen {
+
+    public private(set) var app: XCUIApplication!
+
+    public init() {
+        app = XCUIApplication()
+    }
+
+    public func openUniversalLinkFromRemindersApp(link: String) {
+
+        let reminders = XCUIApplication(bundleIdentifier: "com.apple.reminders")
+        reminders.launch()
+
+        // To dismiss the welcome screen if displayed
+        let continueButton = reminders.buttons["Continue"]
+        if continueButton.exists {
+            continueButton.tap()
+        }
+
+        // Add universal link as reminder
+        let remindersTable = reminders.otherElements["RemindersList.ID.RemindersTable"]
+        remindersTable.tap()
```

Comment:
I noticed some flakiness here (in fact, the first execution of this test on CI for iPad [failed](https://buildkite.com/automattic/woocommerce-ios/builds/12338#018755a5-c8dd-4b40-a1cd-b780b3cf7268/559-3260), but the test passed after a rerun). It fails locally in more than 50% of cases after loading a fresh Simulator.

The fail happens because of the tap. After the very first launch of this app (locally reproducible with `Erase all content and settings` on Simulator), after dismissing all the alerts, the text field might already be shown and focused, and the tap removes both the focus and the TextField from the screen:

https://user-images.githubusercontent.com/73365754/230347235-c05d09a3-3180-4b99-8019-8dcc67ef1536.mov


Possibly not the best solution, but seems to work:

```suggestion
        if !reminders.textFields["Title"].exists{
            remindersTable.tap()
        }
```

Reply:
good catch! yeah looks like it failed on iPad the first time, i didn't repeatedly test on iPad (also didn't clear my local simulator before each run which is probably why it wasn't caught 🙏)

tested this out and proposed change added in https://github.com/woocommerce/woocommerce-ios/pull/9384/commits/72404f3deafa7e47525322778a6802879c7eba9d

Reply:
Thanks for the addition @jostnes! (not blocking in any way, just my feedback: I usually prefer to avoid one lined `if` conditions, because there's no way to set separate breakpoints for the condition and for the conditional code.

Reply:
I looked at this again and found that a few lines above, the `if` wasn't in 1-line. i updated this one to match that and to address your comment: https://github.com/woocommerce/woocommerce-ios/pull/9384/commits/40461d506d80430ac15b1a6c2f6251cc724a2e6c